.github/workflows/rust.yml

```
@@ -312,11 +312,24 @@ jobs:
     steps:
       - name: "Checkout repo"
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # we need full history for cargo semver-checks
       - name: "Select toolchain"
         uses: dtolnay/rust-toolchain@v1
         with:
           toolchain: ${{ needs.Prepare.outputs.nightly_version }}
-      - name: "Install cargo-public-api"
-        run: cargo install cargo-public-api@0.35.0 --locked
-      - name: "Run API checker script"
-        run: ./contrib/check-for-api-changes.sh
+      - name: "Install cargo-semver-checks"
+        run: cargo install cargo-semver-checks --locked
```

Depending on the threat model, we can speed this up with:

```suggestion
      - name: "Install cargo-binstall"
        uses: cargo-bins/cargo-binstall@main
      - name: "Binstall cargo-semver-checks"
        run: cargo binstall cargo-semver-checks --no-confirm
```

---

I think it's fine. We're not checking the signatures/hashes of the source either.